freepbx/Containerfile

```diff
@@ -1,169 +1,409 @@
-FROM docker.io/library/php:7.4-apache as production
+# Stage 1: Build Asterisk
+FROM docker.io/debian:11 as builder
+
+# Set fixed Asterisk version
+ARG ASTERISK_VERSION=18.21.0
+ARG PJSIP_VERSION=2.13.1
+ARG DEBIAN_FRONTEND=noninteractive
+ARG BUILDROOT=/usr/src/asterisk/build
+ARG LIBDIR=/usr/lib
+ARG DATADIR=/var/lib
+ARG MAKEARGS="DEBUG= OPTIMIZE= DESTDIR=${BUILDROOT} ASTVARRUNDIR=/var/run/asterisk ASTDATADIR=${DATADIR}/asterisk ASTVARLIBDIR=${DATADIR}/asterisk ASTDBDIR=${DATADIR}/asterisk/db NOISY_BUILD=1"
+COPY usr/src/asterisk /usr/src/asterisk
+RUN apt-get update && \
```

Comment:
Why did you put all the commands into a single `RUN`? In this way, every time you change something the layer will be discarded and regenerated, and the cache will not be used.

Reply:
To reduce the number of the layers.
When this is in production, I expect that we will change this only on Asterisk version update.

Reply:
> Ok, but this is a multistage container and is not the builder container that will be shipped.

I've added a RUN after the apt-get